src/discordmessageprocessor.ts

```diff
@@ -279,18 +281,48 @@ export class DiscordMessageProcessor {
                     `Could not insert emoji ${id} for msg ${msg.id} in guild ${msg.guild.id}: ${ex}`,
                 );
                 if (html) {
-                    replace = `&lt;${animated ? "a" : ""}:${name}:${id}&gt;`;
+                    replace = `&lt;${animated ? "a" : ""}:${escapeHtml(name)}:${id}&gt;`;
                 } else {
                     replace = `<${animated ? "a" : ""}:${name}:${id}>`;
                 }
             }
-            content = content.replace(results[0],
-                replace);
+            content = content.replace(results[0], replace);
             results = MXC_INSERT_REGEX.exec(content);
         }
         return content;
     }
 
+    public async InsertChannelPills(content: string, msg: Discord.Message, html: boolean = false): Promise<string> {
+        let results = CHANNEL_INSERT_REGEX.exec(content);
+        while (results !== null) {
+            const id = results[ID_CHANNEL_INSERT_REGEX];
+            let replace = "";
+            const channel = msg.guild.channels.get(id);
+            if (channel) {
+                const name = "#" + channel.name;
+                let alias = `#_discord_${msg.guild.id}_${id}:${this.opts.domain}`;
+                try {
+                    // we don't bother here much of checking if objects exists, as the error is catched
+                    // and we use the default alias then, as we wanted to anyways
+                    const matrixRoom = (await this.opts.bot!.ChannelSyncroniser.GetRoomIdsFromChannel(channel))[0];
```

Comment:
If this first roomid is to fail, do we retry with other roomids?

Reply:
nope, should that be added?

Reply:
Also, this should move into channel sync or something anyways as it could be needed for #431 and stuffs

Reply:
Yeah, soru will re-write some things here also prepping for #397

Reply:
kk